traversal/src/main/scala/overflowdb/traversal/RepeatStep.scala

```diff
@@ -0,0 +1,117 @@
+package overflowdb.traversal
+
+import scala.collection.{Iterator, mutable}
+
+object RepeatStep {
+
+  /**
+   * Depth first repeat step implementation
+   * https://en.wikipedia.org/wiki/Depth-first_search
+   *
+   * Given the following example graph:
+   * L3 <- L2 <- L1 <- Center -> R1 -> R2 -> R3 -> R4
+   * the traversal
+   * {{{ center.repeat(_.out).iterate }}}
+   * will iterate the nodes in the following order:
+   * Center, L1, L2, L3, R1, R2, R3, R4
+   *
+   * See RepeatTraversalTests for more detail (and a test for the above).
+   *
+   * Note re implementation: using recursion results in nicer code, but uses the JVM stack, which only has enough space for
+   * ~10k steps. So instead, this uses a programmatic Stack which is semantically identical.
+   * The RepeatTraversalTests cover this case.
+   */
+  object DepthFirst {
+    case class StackItem[A](traversal: Traversal[A], depth: Int)
+
+    def apply[B](repeatTraversal: B => Traversal[B], behaviour: RepeatBehaviour[B]): B => Traversal[B] = {
```

Comment:
I am kind of surprised to see two completely different implementations for DepthFirst and BreadthFirst traversal. I would have expected the difference to only be the collection type for the "worklist" which would be a stack for depth first and a FIFO for breadth first.

Reply:
Yes good point, this is mostly an artifact of my journey to the final version. Initially the two versions looked very similar, but the DFS version used recursion, i.e. the worklist stack was the JVM call stack. Until I realised that that's not a good idea because it quickly blows up. 
Will TAL tomorrow to see if it's worthwhile to bring the two impls closer together. Thanks!

Reply:
yup you're right. i'll still merge this as is to have this version archived in the git history, will send a follow up pr then to unify the two options